src/test/groovy/unit/org/pih/warehouse/api/ProductApiTest.groovy

```diff
@@ -0,0 +1,178 @@
+package unit.org.pih.warehouse.api
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import groovy.transform.builder.Builder
+import io.restassured.RestAssured
+import io.restassured.builder.RequestSpecBuilder
+import io.restassured.http.ContentType
+import io.restassured.http.Cookie
+import io.restassured.path.json.JsonPath
+import io.restassured.specification.RequestSpecification
+import lombok.AllArgsConstructor
+import lombok.Data
+import lombok.NoArgsConstructor
+import org.grails.web.json.JSONObject
+import spock.lang.Shared
+import spock.lang.Specification
+import static io.restassured.RestAssured.*
+import io.restassured.response.Response
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.is
+import static org.hamcrest.Matchers.notNullValue
+import static org.hamcrest.Matchers.lessThanOrEqualTo
+
+class ProductApiTest extends Specification {
+
+    @Shared
+    String baseURI = System.getProperty("server.host")?:"http://localhost"
+
+    @Shared
+    String basePath = System.getProperty("server.path")?:"/openboxes"
+
+    @Shared
+    int port = System.getProperty("server.port")?Integer.valueOf(System.getProperty("server.port")):8080
+
+    @Shared
+    String username = System.getProperty("username")?:"admin"
+
+    @Shared
+    String password = System.getProperty("password")?:"password"
+
+    @Shared
+    String productId
+
+    @Shared
+    RequestSpecification requestSpecification
+
+    // Executed before the first spock test is executed
+    void setupSpec() {
+
+        RestAssured.basePath = basePath
+        RestAssured.baseURI = baseURI
+        RestAssured.port = port
+
+        JSONObject jsonObject = new JSONObject().
+                put("username", username).
+                put("password", password).
+                put("location", 1)
+
+        Response response = given().log().all().
+                contentType(ContentType.JSON).
+                body(jsonObject.toString()).
+                post("/api/login")
```

Comment:
here dots also should be at the beginning of the line

Reply:
Again, where have you seen this syntax? Is that a javascript thing?